src/payment/strategies/amazon-pay/amazon-pay-payment-strategy.ts

```diff
@@ -210,4 +220,38 @@ export default class AmazonPayPaymentStrategy implements PaymentStrategy {
             })
         );
     }
+
+    private _processPaymentWith3ds(sellerId: string, referenceId: string, methodId: string, useStoreCredit: boolean, options: PaymentRequestOptions): Promise<InternalCheckoutSelectors> {
+        return new Promise((resolve, reject) => {
+            if (!this._window.OffAmazonPayments) {
+                return Promise.reject(new NotInitializedError(NotInitializedErrorType.PaymentNotInitialized));
+            }
+
+            return this._window.OffAmazonPayments.initConfirmationFlow(
+            sellerId,
+            referenceId,
+            (confirmationFlow: AmazonPayConfirmationFlow) => {
+                return this._store.dispatch(
+                    this._orderActionCreator.submitOrder({useStoreCredit}, options)
+                )
+                    .then(() => this._store.dispatch(
+                        this._remoteCheckoutActionCreator.initializePayment(methodId, {
+                            referenceId,
+                            useStoreCredit,
+                        }))
+                    )
+                    .then(() => {
+                        confirmationFlow.success();
+
+                        return new Promise<never>(() => {});
```

Comment:
I think is a little bit weird to use this type of Promise, and actually your global promise is getting a `resolve` scope, so I think you need to use this scope to resolve you Promise, maybe you can simply use a `resolve()`;

Reply:
I don't need the promise to be resolved in this case because I'll do a redirect and amazonpay will handle this so if I resolve this promise the redirect won't works